codec-http2/src/main/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributor.java

```diff
@@ -211,19 +366,108 @@ int streamableBytes0(Http2Stream stream) {
     }
 
     /**
+     * For testing only!
+     */
+    boolean isChild(int childId, int parentId, short weight) {
+        State parent = state(parentId);
+        State child;
+        return parent.children.containsKey(childId) &&
+                (child = state(childId)).parent == parent && child.weight == weight;
+    }
+
+    /**
+     * For testing only!
+     */
+    int numChildren(int streamId) {
+        State state = state(streamId);
+        return state == null ? 0 : state.children.size();
+    }
+
+    /**
+     * Notify all listeners of the priority tree change events (in ascending order)
+     * @param events The events (top down order) which have changed
+     */
+    void notifyParentChanged(List<ParentChangedEvent> events) {
```

Comment:
This is internal and the intention is to be able to avoid garbage created by an iterator bcz I know this is backed by an ArrayList in practice.